web/proxy.js

```diff
@@ -59,6 +59,15 @@ const server = http.createServer((request, response) => {
 
     request.pipe(proxyRequest);
     proxyRequest.on('response', (proxyResponse) => {
+        // Get the existing CSP header value
+        let existingCSP = proxyResponse.headers['content-security-policy'];
+
+        // Append newDot URLs in the frame-ancestors CSP headers to allow the oldDot to be embedded in the Expensify App
+        existingCSP += ' https://staging.new.expensify.com';
```

Comment:
A couple of questions here:
1. Currently we just have a single directive `frame-ancestors` that's why I just appended the staging URL to it. And this is what the response looks like
<img width="1724" alt="Screenshot 2023-09-13 at 1 26 18 PM" src="https://github.com/Expensify/App/assets/35863227/46c9884d-38b9-46c0-8907-0a53b5e2c0e7">

But this would break if there are multiple directives as I am just appending to the end.

2. Using a proxy server we can just whitelist localhost:8082, right? For staging(staging.new.expensify.com) I think we will still need to update the Salt file?

Reply:
I think we can update the Salt file to include the staging URL and in the proxy file we can update the headers to include localhost URL?